File: USACO/2019/December/Silver/moobuzz.py
#Doesn't work
fin=open("moobuzz.in")
N=int(fin.readline().rstrip())
fin.close()

def getAns(N):
  return N-((N-N%3)/3) - ((N-N%5)/5) + ((N-N%15)/15)


def search(small, big, goal):
  mid=(small +big)//2
  if getAns(mid)==goal:
    return mid
  elif getAns(mid)<N:
    return search(mid, big, N)
  else:
    return search(small, mid, N)

r=int(search(0, 10E9, N))
with open("moobuzz.out", "w") as fout:
  fout.write(str(r)+"\n")  

File: USACO/2019/December/Silver/moobuzz.cpp
#include <iostream>
#include <fstream>
#include <string>
using namespace std;

//Doesn't work

int main() {
  ifstream myfile;
  myfile.open("moobuzz.in");
  int N;
  myfile >> N;
  myfile.close();
  int realNum=0;
  int c=0;

  while(c!=N){
    if(realNum%3!=0 && realNum%5!=0){
      c+=1;
    }
    realNum+=1;
  }
  realNum-=1;


  // cout << realNum;
  ofstream fout;
  fout.open("moobuzz.out");
  fout << realNum;
  fout << "\n";
  fout.close();
}